test/models/drive_test.dart

```diff
@@ -0,0 +1,489 @@
+              seats: 1)
+        ],
+        createDependencies: false,
+      );
+      final Ride ride = RideFactory().generateFake(
+        startTime: DateTime(2022, 2, 2, 14, 50),
+        endTime: DateTime(2022, 2, 2, 15),
+        seats: 1,
+      );
+      expect(drive.isRidePossible(ride), false);
+    });
+    test('share ride with other rides', () async {
+      final Drive drive = DriveFactory().generateFake(
+        startTime: DateTime(2022, 2, 2, 14),
+        endTime: DateTime(2022, 2, 2, 16),
+        seats: 4,
+        rides: <Ride>[
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14, 20),
+              endTime: DateTime(2022, 2, 2, 15),
+              status: RideStatus.approved,
+              seats: 2),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14, 40),
+              endTime: DateTime(2022, 2, 2, 14, 50),
+              status: RideStatus.approved,
+              seats: 1),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 15),
+              endTime: DateTime(2022, 2, 2, 15, 30),
+              status: RideStatus.approved,
+              seats: 2),
+        ],
+      );
+      final Ride ride = RideFactory().generateFake(
+          startTime: DateTime(2022, 2, 2, 14, 50),
+          endTime: DateTime(2022, 2, 2, 15, 10),
+          status: RideStatus.approved,
+          seats: 1);
+      expect(drive.isRidePossible(ride), true);
+    });
+  });
+  group('Drive.cancel', () {
+    test('cancelled is being changed from false to true', () async {
+      final Drive drive = DriveFactory().generateFake(cancelled: false, createDependencies: false);
+      when.call(driveProcessor.processUrl(any)).thenReturn(drive.toString());
+      drive.cancel();
+      expect(drive.cancelled, true);
+    });
+  });
+  group('Drive.toString', () {
+    test('returns a string representation of the drive', () async {
+      final Drive drive = DriveFactory().generateFake(
+        id: 1,
+        start: 'start',
+        startTime: DateTime.parse('2022-02-02T00:00:00.000Z'),
+        end: 'end',
+        endTime: DateTime.parse('2023-03-03T00:00:00.000Z'),
+        driverId: 5,
+        createDependencies: false,
+      );
+      expect(
+        drive.toString(),
+        'Drive{id: 1, from: start at 2022-02-02 00:00:00.000Z, to: end at 2023-03-03 00:00:00.000Z, by: 5}',
+      );
+    });
+  });
+  group('Drive.equals', () {
+    final Drive drive0 = DriveFactory().generateFake(
+      id: 1,
+      createdAt: DateTime(2022, 10),
+      start: 'Berlin',
+      startPosition: Position(1, 1),
+      startTime: DateTime(2022, 11),
+      end: 'Frankfurt',
+      endPosition: Position(2, 2),
+      endTime: DateTime(2022, 12),
+      seats: 1,
+      cancelled: false,
+      driverId: 2,
+      createDependencies: false,
+    );
+    test('same drive', () async {
+      expect(drive0.equals(drive0), true);
```

Comment:
I think you should use a deepcopy here. That would show that we're using value equality, not reference equality. For Ride too